lib: location: Add multicell_location stuff (#9690)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/location.rst

* :kconfig:option:`CONFIG_NRF_CLOUD_REST` - Uses REST APIs to communicate with `nRF Cloud`_ if :kconfig:option:`CONFIG_NRF_CLOUD_MQTT` is not set.
* :kconfig:option:`CONFIG_NRF_CLOUD_MQTT` - Uses MQTT transport to communicate with `nRF Cloud`_.

> Both cellular and Wi-Fi location services are done externally by the application or selected using the runtime configuration, in which case the available services must be configured first.

Reviewer: ```suggestion
Both cellular and Wi-Fi location services are handled externally by the application or selected using the runtime configuration, in which case you must first configure the available services.
```

#. Set any required non-default values to the structures.
#. Call the :c:func:`location_request` function with the configuration.

> There are different ways of using :c:func:`location_request` function. Below you'll see examples.

Reviewer: ```suggestion
You can use the :c:func:`location_request` function in different ways, as in the following examples.
```
